synapse/storage/databases/main/cache.py

```diff
@@ -226,6 +249,9 @@ def _invalidate_caches_for_event(
         relates_to: Optional[str],
         backfilled: bool,
     ) -> None:
+        # XXX: If you add something to this function make sure you add it to
+        # `_invalidate_caches_for_room` as well.
+
```

Comment:
Is this accurate? You now call explictly call `_invalidate_caches_for_events` and `_invalidate_caches_for_room` in the delete room case.

Reply:
Whoops, that should say `_invalidate_caches_for_events`